### PR TITLE
[luci/pass] Extract canonicalize method

### PR DESCRIPTION
**Pull request by seanshpark**
This will extract canonicalize method with
- run before convert_nchw_to_nhwc
- also run remove dead node pass to remove unnecessary nodes

so that convert_nchw_to_nhwc have have canonicalized properties.

### CODE REVIEW

**Comment by seanshpark** [comment]
- for #13531
- from #13532

File: compiler/luci/pass/src/CircleOptimizer.cpp [hunk 1]
```diff
@@ -217,6 +217,19 @@ CircleOptimizer::Options *CircleOptimizer::options(void)
   return _options.get();
 }
 
+void CircleOptimizer::canonicalize(loco::Graph *g) const
+{
+  logo::Phase phase;
+
+  phase.emplace_back(std::make_unique<luci::CanonicalizePass>());
+  phase.emplace_back(std::make_unique<logo::RemoveDeadNodeWithQueryPass>());
```
Comment (jinevening) [inline compiler/luci/pass/src/CircleOptimizer.cpp:225]: How about adding shape/dtype inference passes? If canonicalization updates dtype/shape, those passes will be needed too.

**Review by shs-park — APPROVED** [review]
LGTM!
=)

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**